Fix for subviews not resizing when using window snapping tool and fixing inconsistent state issue.

jsakuda opened the pull request:
Fixing issue where subview frames ended up in an inconsistent state (this was printing out to the console). Also changing the way resize is detected to fix issues with subviews sometimes not resizing.

raffael commented:
Thanks for the pull request!
I think we then additionally need to introduce a proxy `NSWindowDelegate`, as, e.g., `INAppStoreWindow` does it, in order to still allow an external window delegate. Otherwise, setting a different delegate would break your code?

jsakuda commented:
You are right, sorry slipped my mind. I'll add one and push the code to my master branch.

jsakuda committed:
Replacing resizeSubviews delegate method shouldAdjustSizeOfSubview.

This fixes the issue with subviews being in an inconsistent state.
It also fixes issues with subviews not resizing when the window resizes.

jsakuda commented:
So I originally did add the proxy delegate code. However, I realized that the code had started working without the delegate at some point. Since I had remedied the resizing issues before I fixed the error about the subviews being in an inconsistent state, I didn't realize that just fixing the inconsistent state would solve the resizing issues.

I've modified my commit tree to only have the edits to the NSSplitViewDelegate. Basically just using shouldAdjustSizeOfSubview instead of resizeSubviews fixed it. Ended up being much simpler than before, sorry for all the trouble. Just FYI, that this should fix Issue #1.

raffael commented:
Thanks!
